File: client/src/types/company.ts
export type Company = {
  id: string;
  name: string;
  careerPageUrl?: string;
  websiteUrl?: string;
  location?: string;
  notes?: string;
  isFavorite: boolean;
  isArchived: boolean;
};

export type CreateCompanyRequest = {
  name: string;
  location?: string;
  websiteUrl?: string;
  careerPageUrl?: string;
  notes?: string;
};

export type UpdateCompanyRequest = CreateCompanyRequest;

File: server/JobTracker.API/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace JobTracker.API.Models;

public class Company : BaseEntity
{
    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? CareerPageUrl { get; set; }

    [MaxLength(500)]
    public string? WebsiteUrl { get; set; }

    [MaxLength(500)]
    public string? Location { get; set; }

    [MaxLength(5000)]
    public string? Notes { get; set; }

    public bool IsFavorite { get; set; }
    public bool IsArchived { get; set; }

    public Guid? UserId { get; set; }
    public User? User { get; set; }

    public ICollection<JobApplication> JobApplications = new List<JobApplication>();
}